### PR TITLE
stop action creator plus minor refactoring

### PR DESCRIPTION
**Pull request by blimpich**

### CODE REVIEW

File: spec/redux/actionCreators-test.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ describe("The actionCreators'", () => {
     });
 
     it("dispatches a START_LOAD_RUNTIME action first", () => {
-      expect(store.getActions()[0]).toEqual({type: "START_LOAD_RUNTIME"});
+\      expect(store.getActions()[0]).toEqual({type: "START_LOAD_RUNTIME"});
```
Comment (pcardune) [inline spec/redux/actionCreators-test.js]: derp

File: src/stubCompiler.js [hunk 2]
```diff
@@ -107,5 +107,9 @@ export default {
       var result = bytecode();
       window.setTimeout(() => resolve(result), 1000);
     });
+  },
+
+  stop() {
+    new Error("stop");
```
Comment (pcardune) [inline src/stubCompiler.js]: better to say `console.warn("Not Implemented yet")`

**Comment by blimpich** [comment]
sorry for lumping all these minor commits together, normally I would try to separate these out into separate pull requests

File: src/redux/actionCreators.js [hunk 3]
```diff
@@ -30,16 +30,25 @@ export function run(src) {
     runtimeApi
       .parse(src)
       .then(ast => {
+        if (!getState().runCode.stage) {
```
Comment (pcardune) [inline src/redux/actionCreators.js]: Can you update this to use the new selectors?

**Comment by pcardune** [comment]
one small change and then I will merge. btw, feel free to switch the label for a pull request to "needs review" when you are ready for me to take another look.

File: src/components/Stop.js [hunk 4]
```diff
@@ -26,7 +32,7 @@ Stop.propTypes = {
 
 export default connect(
   state => ({
-    running: state.running
+    running: Object.values(constants.runtimeStages).includes(state.runCode.stage),
```
Comment (pcardune) [inline src/components/Stop.js]: shouldn't this be `selectors.isRunning(state)`?

**Pull request merged**